### PR TITLE
Standardized Testing Module

### PR DESCRIPTION
**Pull request by hyang0129**
See #992 

This refactors a previous commit to make the discover classes function accessible to other modules for the creation of future standardized testing of classes within a module

### CODE REVIEW

**Comment by hyang0129** [comment]
@seanpmorgan please review the refactoring

File: tensorflow_addons/optimizers/tests/standard_tests.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 import tensorflow as tf
```
Comment (WindQAQ) [inline tensorflow_addons/optimizers/tests/standard_tests.py:18]: Why the file name is changed?

**Review by WindQAQ — COMMENTED** [review]
Sorry for the late review. It's a good idea to me. One minor question that I want to clarify. Thank you!

File: tensorflow_addons/optimizers/tests/standard_tests.py (hunk 1 above)
Comment (hyang0129) [reply tensorflow_addons/optimizers/tests/standard_tests.py:18]: I think I changed it because there would be multiple tests? We can change it back?

Comment (WindQAQ) [reply tensorflow_addons/optimizers/tests/standard_tests.py:18]: Let's change it back. I think it's better to have a standard naming convention for file!

**Review by WindQAQ — APPROVED** [review]
Thank you!

**Pull request merged**